Assets/Scripts/Wizzard/WizzardController.cs

```diff
@@ -35,28 +36,23 @@ void Update()
     {
         if (InputEnabled)
         {
-            GameObject fireball = null;
             if (Input.GetKeyDown(KeyCode.F))
             {
-                fireball = (GameObject)Instantiate<GameObject>(FireballBallistic);
-            }
-            else if (Input.GetKeyDown(KeyCode.G))
-            {
-                fireball = (GameObject)Instantiate<GameObject>(FireballStraight);
-            }
-            else if (Input.GetKeyDown(KeyCode.H))
-            {
-                fireball = (GameObject)Instantiate<GameObject>(FireballTime);
-            }
+                if (wizardStats.Spells.Count > 0)
+                {
+                    GameObject fireball = SpellFactory.GetSpell(wizardStats.Spells[currentSpell]);
 
-            if (fireball != null)
-            {
-                var x = FireStrength * Mathf.Cos(Angle * Mathf.Deg2Rad);
-                var y = FireStrength * Mathf.Sin(Angle * Mathf.Deg2Rad);
-                int directionNum = facingRight ? 1 : -1;
-                fireball.transform.position = this.transform.position + new Vector3(directionNum * 0.4f, 0.4f, 0f);
-                var rigidBodyFireball = fireball.GetComponent<Rigidbody2D>();
-                rigidBodyFireball.AddForce(new Vector2(x, y));
+                    if (fireball != null)
+                    {
+                        Spell spell = fireball.GetComponent<Spell>();
+                        spell.Activate();
+                        wizardStats.Spells.Remove(wizardStats.Spells[currentSpell]);
```

Comment:
Troche ta tablica speli jest dziwna. Spelle maja byc pokazywane na zasadzie A-spell1 B spell2 przykladowo ABBBBBAAAABBB zamiast A szt 5 B szt 8 Nie bedzie to zbyt czytelne dla uzytkownika chyba ze taki byl Twoj zamiar?